onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1648,6 +1648,23 @@ Example 4:
                 output_counts = [1, 2, 2, 1]
               )DOC");
 
+  //see:https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html
+  ONNX_CONTRIB_OPERATOR_SCHEMA(CDist)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Attr("metric",
+            "The distance metric to use. If a string, the distance function can be \"braycurtis\", \"canberra\", "
```

Comment:
I think we should reduce the number of supported distances to begin with. It would be greater to support Minkowski for the nearest neighbors but it requires an additional parameter p (integer).